invenio_rdm_records/resources/deserializers/rocrate/schema.py

```diff
@@ -0,0 +1,106 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2022 CERN.
+#
+# Invenio-RDM-Records is free software; you can redistribute it and/or modify
+# it under the terms of the MIT License; see LICENSE file for more details.
+
+"""RO-Crate schema."""
+
+from marshmallow import EXCLUDE, Schema, fields, pre_load
+from marshmallow_utils.fields import SanitizedHTML, SanitizedUnicode
+
+
+def _list_value(lst):
+    """Return the first value from single-value lists."""
+    if isinstance(lst, list) and len(lst) == 1:
+        return lst[0]
+    return lst
+
+
+class ROCrateSchema(Schema):
+    """Schema for the loading RO-Crate into record metadata."""
+
+    class Meta:
+        """Meta attributes for the schema."""
+
+        unknown = EXCLUDE
+
+    LIST_VALUE_FIELDS = [
+        "license",
+        "description",
+        "datePublished",
+    ]
+
+    @pre_load
+    def list_to_value(self, data, **kwargs):
+        """Extract top-level single-item list field values."""
+        for field in self.LIST_VALUE_FIELDS:
+            if field in data:
+                data[field] = _list_value(data[field])
+        return data
+
+    resource_type = fields.Constant({"id": "dataset"})
+    title = SanitizedUnicode(data_key="name", required=True)
+    description = SanitizedHTML()
+    publication_date = fields.Method(
+        deserialize="load_publication_date",
+        data_key="datePublished",
+    )
+
+    def load_publication_date(self, value):
+        """Load publication date."""
+        if value:
+            return value[:10]
+
+    creators = fields.Method(
+        deserialize="load_creators",
+        required=True,
+        data_key="author",
+    )
+
+    def load_creators(self, value):
+        """Load creators."""
+        if value and isinstance(value, list):
+            creators = []
+            for obj in value:
+                person_or_org = {}
+                affiliations = []
+                if obj.get("@type") == "Person":
+                    person_or_org["type"] = "personal"
+                    person_or_org["family_name"] = _list_value(obj.get("familyName"))
+                    person_or_org["given_name"] = _list_value(obj.get("givenName"))
+                    if obj.get("affiliation"):
+                        for a in obj.get("affiliation"):
+                            affiliations.append({"name": a.get("name")})
+                elif obj.get("@type") == "Organization":
+                    person_or_org["type"] = "organizational"
+                    person_or_org["name"] = obj.get("name")
+
+                creators.append(
+                    {
+                        "person_or_org": person_or_org,
+                        "affiliations": affiliations,
+                    }
+                )
+            return creators
+
+    rights = fields.Method(deserialize="load_rights", data_key="license")
+
+    def load_rights(self, value):
+        """Load rights."""
+        if value:
```

Comment:
minor: In this case, since it is not required there's no need for a `ValidationError`. However, I wonder if the `if` is needed. Is the function actually called if the `data_key` is not present?